backend/infrahub/core/schema/attribute_schema.py

```diff
@@ -53,16 +63,36 @@ def kind_options(cls, v: str) -> str:
 
     @model_validator(mode="before")
     @classmethod
-    def validate_dropdown_choices(cls, values: dict[str, Any]) -> dict[str, Any]:
+    def validate_dropdown_choices(cls, values: Any) -> Any:
         """Validate that choices are defined for a dropdown but not for other kinds."""
-        if values.get("kind") != "Dropdown" and values.get("choices"):
+        if isinstance(values, dict):
+            kind = values.get("kind")
+            choices = values.get("choices")
+        elif isinstance(values, AttributeSchema):
+            kind = values.kind
+            choices = values.choices
+        else:
+            return values
+        if kind != "Dropdown" and choices:
             raise ValueError(f"Can only specify 'choices' for kind=Dropdown: {values['kind']}")
 
-        if values.get("kind") == "Dropdown" and not values.get("choices"):
+        if kind == "Dropdown" and not choices:
```

Comment:
Nit pick: I am wondering if we have a global constant or somehow a reference to this `"Dropdown"`

Reply:
I don't think we should use something like this from the SDK. At some point we'll end up with an issue where we're missing a new type from the SDK so we need to add it there first. Later I think we should have this within Infrahub core.